express-notes-api/server.js

```diff
@@ -1,20 +1,19 @@
 import express from 'express';
-import * as fs from 'node:fs';
 import * as fsPromises from 'node:fs/promises';
 
 // const fileData = fs.readFileSync('data.json', 'utf8');
 // const data = JSON.parse(fileData);
 // const notes = data.notes;
-// const writeFileDataJSON = () => fsPromises.writeFile('data.json', JSON.stringify(data, null, 2));
+const writeFileDataJSON = (data) => fsPromises.writeFile('data.json', JSON.stringify(data, null, 2));
 
-const getFunction = (req, res) => {
-  const fileData = fs.readFileSync('data.json', 'utf8');
+const getFunction = async (req, res) => {
+  const fileData = await fsPromises.readFile('data.json', 'utf8');
```

Comment:
Since you do this read & parse several times, you could put them into a function, just like you did writeFileDataJSON.